main doesn't compile on Xcode beta 5

### Expected behavior
Latest `main` compiles and runs tests fine with latest Xcode 14 Seed.

### Actual behavior
```
Building for debugging...
/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/module.modulemap:2:12: error: header '/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h' not found
    header "/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h"
           ^
/Users/davidnadoba/Repositories/swift-cluster-membership/Tests/SWIMTestKit/TestMetrics.swift:32:18: error: could not build Objective-C module 'SWIM'
@testable import SWIM
                 ^
/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/module.modulemap:2:12: error: header '/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h' not found
    header "/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h"
           ^
/Users/davidnadoba/Repositories/swift-cluster-membership/Tests/SWIMTestKit/TestMetrics.swift:32:18: error: could not build Objective-C module 'SWIM'
@testable import SWIM
                 ^
/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/module.modulemap:2:12: error: header '/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h' not found
    header "/Users/davidnadoba/Repositories/swift-cluster-membership/.build/arm64-apple-macosx/debug/SWIM.build/SWIM-Swift.h"
           ^
/Users/davidnadoba/Repositories/swift-cluster-membership/Tests/SWIMTestKit/TestMetrics.swift:32:18: error: could not build Objective-C module 'SWIM'
@testable import SWIM
                 ^

error: fatalError
```
### Steps to reproduce
```
git clone git@github.com:apple/swift-cluster-membership.git
cd swift-cluster-membership
swift test
```

### Version/commit hash
latest `main`
`68b2da6afe4491fca3be3f1265a1ee2697f33ff3`

### Swift & OS version (output of `swift --version && uname -a`)
Xcode: Version 14.0 beta 5 (14A5294e)
```
swift-driver version: 1.62.3 Apple Swift version 5.7 (swiftlang-5.7.0.123.8 clang-1400.0.29.50)
Target: arm64-apple-macosx13.0
Darwin Davids-MBP.fritz.box 22.0.0 Darwin Kernel Version 22.0.0: Wed Jul 20 01:53:07 PDT 2022; root:xnu-8792.0.188.141.3~2/DEVELOPMENT_ARM64_T6000 arm64
```

--- a/Package.swift
+++ b/Package.swift
@@ -93,7 +93,7 @@ var targets: [PackageDescription.Target] = [
     .testTarget(
         name: "SWIMTestKit",
         dependencies: [
-//            "SWIM",
+            "SWIM",
             .product(name: "NIO", package: "swift-nio"),
             .product(name: "Logging", package: "swift-log"),
             .product(name: "Metrics", package: "swift-metrics"),

--- a/Package.swift
+++ b/Package.swift
@@ -93,6 +93,7 @@ var targets: [PackageDescription.Target] = [
     .testTarget(
         name: "SWIMTestKit",
         dependencies: [
+//            "SWIM",
             .product(name: "NIO", package: "swift-nio"),
             .product(name: "Logging", package: "swift-log"),
             .product(name: "Metrics", package: "swift-metrics"),

--- a/Sources/ClusterMembership/Node.swift
+++ b/Sources/ClusterMembership/Node.swift
@@ -64,10 +64,10 @@ public struct Node: Hashable, Sendable, Comparable, CustomStringConvertible {
     }
 }
 
-extension Node {
+public extension Node {
     // Silly but good enough comparison for deciding "who is lower node"
     // as we only use those for "tie-breakers" any ordering is fine to be honest here.
-    public static func < (lhs: Node, rhs: Node) -> Bool {
+    static func < (lhs: Node, rhs: Node) -> Bool {
         if lhs.protocol == rhs.protocol, lhs.host == rhs.host {
             if lhs.port == rhs.port {
                 return (lhs.uid ?? 0) < (rhs.uid ?? 0)

--- a/docker/docker-compose.2004.main.yaml
+++ b/docker/docker-compose.2004.main.yaml
@@ -10,6 +10,8 @@ services:
 
   unit-tests:
     image: swift-cluster-membership:20.04-main
+    environment:
+      - EXPLICIT_TARGET_DEPENDENCY_IMPORT_CHECK=--explicit-target-dependency-import-check error
 
   unit-tests-until-failure:
     image: swift-cluster-membership:20.04-main
@@ -19,6 +21,8 @@ services:
 
   test:
     image: swift-cluster-membership:20.04-main
+    environment:
+      - EXPLICIT_TARGET_DEPENDENCY_IMPORT_CHECK=--explicit-target-dependency-import-check error
 
   bench:
     image: swift-cluster-membership:20.04-main

--- a/docker/docker-compose.yaml
+++ b/docker/docker-compose.yaml
@@ -42,7 +42,7 @@ services:
 
   test:
     <<: *common
-    command: /bin/bash -cl "swift test --enable-test-discovery -Xswiftc -DTESTS_LEAKS && ./scripts/integration_tests.sh"
+    command: /bin/bash -cl "swift test $${EXPLICIT_TARGET_DEPENDENCY_IMPORT_CHECK-} --enable-test-discovery -Xswiftc -DTESTS_LEAKS && ./scripts/integration_tests.sh"
 #    command: /bin/bash -cl "WARNINGS_AS_ERRORS=yes swift test --enable-test-discovery -Xswiftc -DTESTS_LEAKS && ./scripts/integration_tests.sh"
 
   # util

